src/category_theory/category/ulift.lean

```diff
@@ -55,4 +56,57 @@ def ulift.equivalence : C ≌ (ulift.{u2} C) :=
   inv_hom_id' := by {ext, change (𝟙 _) ≫ (𝟙 _) = 𝟙 _, simp} },
   functor_unit_iso_comp' := λ X, by {change (𝟙 X) ≫ (𝟙 X) = 𝟙 X, simp} }
 
+instance [is_filtered C] : is_filtered (ulift.{u2} C) :=
+{ cocone_objs := λ X Y, ⟨⟨is_filtered.max X.down Y.down⟩,
+    is_filtered.left_to_max _ _, is_filtered.right_to_max _ _, trivial⟩,
+  cocone_maps := λ X Y f g, ⟨⟨is_filtered.coeq f g⟩,
+    is_filtered.coeq_hom _ _, is_filtered.coeq_condition _ _⟩,
+  nonempty := ⟨⟨is_filtered.nonempty.some⟩⟩ }
+
+instance [is_cofiltered C] : is_cofiltered (ulift.{u2} C) :=
+{ cocone_objs := λ X Y, ⟨⟨is_cofiltered.min X.down Y.down⟩,
+    is_cofiltered.min_to_left _ _ , is_cofiltered.min_to_right _ _, trivial⟩,
+  cocone_maps := λ X Y f g, ⟨⟨is_cofiltered.eq f g⟩,
+    is_cofiltered.eq_hom _ _, is_cofiltered.eq_condition _ _⟩,
+  nonempty := ⟨⟨is_cofiltered.nonempty.some⟩⟩ }
+
+variable (C)
+/-- `as_small C` is a small category equivalent to `C`.-/
+@[nolint unused_arguments]
+def as_small := ulift.{max v u1} C
+variable {C}
+
+instance : small_category (as_small C) :=
+{ hom := λ X Y, ulift $ X.down ⟶ Y.down,
+  id := λ X, ⟨𝟙 _⟩,
+  comp := λ X Y Z f g, ⟨f.down ≫ g.down⟩ }
+
+/-- One half of the equivalence between `C` and `as_small C`. -/
+@[simps]
+def as_small.up : C ⥤ as_small C :=
+{ obj := λ X, ⟨X⟩,
+  map := λ X Y f, ⟨f⟩ }
+
+/-- One half of the equivalence between `C` and `as_small C`. -/
+@[simps]
+def as_small.down : as_small C ⥤ C :=
+{ obj := λ X, X.down,
+  map := λ X Y f, f.down }
+
+/-- The equivalence between `C` and `as_small C`. -/
+@[simps]
+def as_small.equiv : C ≌ as_small C :=
```

Comment:
This gives the equivalence of a category to a particular `ulift` of it where the argument is the universe level of the morphisms - but any category should be equivalent to any `ulift` of itself

Reply:
This can be simulated with a combination of `ulift` and `as_small`, but I agree that a direct construction could be better. I'm a bit worried that dealing with three universes will get annoying quickly, but I'll have a go at generalizing later today.

Reply:
I'm even more confused then - what's the purpose of `as_small` when we already have `ulift`? It's just a special case, so you can just make it an abbreviation instead of making the new category instances and functors on it - as far as I can see `as_small.up` is the same thing as `ct.ulift.up`. Am I missing something?

Reply:
The category instance on `ulift C` doesn't change the universe level of morphisms.

Reply:
Ahh I see, so `ulift C` lifts the objects up, and `as_small` lifts the morphisms up to the objects. In that case I suspect the right option is to change `as_small` to `ulift_hom` or something like that, which only changes the universe level of morphisms but to any level (rather than just that of the objects), and then if you need to change both you can compose? Or am I misunderstanding `as_small` once again?

Reply:
It's a composition of `ulift` and `ulift_hom` for convenience.  It should be possible to factor `as_small`.

Reply:
I tried making a `ulift_hom`, but I don't see how to make such a construction actually useful. Namely, I don't see how to add a universe parameter to a type alias so that the typeclass system is able to deduce which universe level to apply to the `ulift` in the hom-types.